src/SignalException.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Amp;
+
+/**
+ * Used as the previous exception to {@see CancelledException} when a {@see SignalCancellation} is triggered.
+ *
+ * @see SignalCancellation
+ */
+class SignalException extends \Exception
```

Comment:
Should we have another exception for that?

Reply:
It can be nice if used with CombinedCancellation to differentiate which cancellation was used. It's not strictly necessary, though mirrors TimeoutException with TimeoutCancellation.